packages/playwright-test/src/runner.ts

```diff
@@ -237,24 +237,34 @@ export class Runner {
     return projects;
   }
 
-  private async _collectFiles(projects: FullProjectInternal[], testFileFilter: Matcher): Promise<Map<FullProjectInternal, string[]>> {
-    const files = new Map<FullProjectInternal, string[]>();
+  private async _collectFiles(projects: FullProjectInternal[], testFileFilter: Matcher): Promise<{filesByProject: Map<FullProjectInternal, string[]>; setupFiles: Set<string>}> {
+    const extensions = ['.js', '.ts', '.mjs', '.tsx', '.jsx'];
+    const testFileExtension = (file: string) => extensions.includes(path.extname(file));
+    const filesByProject = new Map<FullProjectInternal, string[]>();
+    const setupFiles = new Set<string>();
     for (const project of projects) {
       const allFiles = await collectFiles(project.testDir, project._respectGitIgnore);
+      const setupMatch = createFileMatcher(project._setup);
```

Comment:
Will this match everything if there is no `project.setup` defined?

Reply:
No, it is always false for empty filter.